samples/model-builder/model_registry/upload_new_default_model_version_sample.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START aiplatform_model_registry_upload_new_default_model_version_sample]
+
+from google.cloud import aiplatform
+
+
+def upload_new_default_model_version_sample(
+    model_id: str,
+    artifact_uri: str,
+    serving_container_image: str,
+    project: str,
+    location: str,
+):
+    """
+    Uploads a new default version of a model with ID 'model_id'.
+    Args:
+        model_id: The ID of the model to upload a new version to. Parent resource name of the model is also accepted.
```

Comment:
maybe we should rename this to parent_model?

because we can also specify a user defined model id to replace the numeric model id, in API proto, that field is named model_id

And maybe we should have a sample for uploading a new model with user defined id?

Reply:
Oh I see the point @deanchen-google. I will correct the sample accordingly. About uploading a new model with user defined id we already have [the official example](https://cloud.google.com/vertex-ai/docs/model-registry/import-model#aiplatform_upload_model_sample-python).

Reply:
@deanchen-google @samgoodman all reviews are in place. Samples are ready to be merged into main.